superset-frontend/src/dashboard/actions/dashboardState.js

```diff
@@ -315,6 +316,9 @@ export function saveDashboardRequest(data, id, saveType) {
         ? undefined
         : ensureIsArray(roles).map(r => (hasId(r) ? r.id : r)),
       slug: slug || null,
+      tags: !isFeatureEnabled(FeatureFlag.TaggingSystem)
+        ? undefined
+        : ensureIsArray(tags).map(r => (hasId(r) ? r.id : r)),
```

Comment:
Is there any chance `tags` could be undefined here? If so, you could add `ensureIsArray(tags || [])`